File: numinlist/num_in_list.ts
export const loop = (list: number[], num: number) => {
  for (let n of list) {
    if (n === num) {
      return true;
    }
  }
  return false;
};

export const std = (list: number[], num: number) => {
  return list.includes(num);
};

File: elixir/lib/polyglot/num_in_list.ex
# https://github.com/joncalhoun/algorithmswithgo.com/tree/master/module01#05---determine-if-a-number-is-in-a-list-code

defmodule Polyglot.NumInList do
  def pattern_match(nil, _), do: false
  def pattern_match([], _), do: false

  def pattern_match(list, num) do
    [head | tail] = list
    head == num || pattern_match(tail, num)
  end

  def std(list, num) do
    Enum.member?(list, num)
  end
end
